data_override/data_override.F90

```diff
@@ -117,56 +66,18 @@ module data_override_mod
      module procedure data_override_UG_2d
 end interface
 
+integer :: atm_mode = 0
+integer :: ocn_mode = 0
+integer :: lnd_mode = 0
+integer :: ice_mode = 0
```

Comment:
I added the `class(*)` dispatcher subroutines because I didn't realize that `module procedure x` could be used when `x` is a `use`'d subroutine (I thought `module procedure` was only for procedures in the module's `contains` section). But it actually seems to work fine, so the `class(*)` subroutines have been replaced by interfaces in b89c8db621164eef22e0d9220a2046a4a1ceaa98.

Reply:
`atm_mode`, `ocn_mode`, `lnd_mode`, and `ice_mode` all take on values `0` (uninitialized), `4` (r4 mode), and `8` (r8 mode). This will be clarified with doxygen comments.

Reply:
Comments added in eb0137c3ee6b3804dea57984bb670168cff9f061.